tests/functional/test_plugin_auth.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (C) 2017  Custodia Project Contributors - see LICENSE file
+
+from __future__ import absolute_import
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+
+import pytest
+
+from .base import AuthPlugin, CustodiaServer, CustodiaTestEnvironment
+
+
+class TestBasicsAuthPlugins(CustodiaTestEnvironment):
+    @pytest.mark.parametrize("meta_uid,meta_gid,expected_access", [
+        ('correct_id', 'correct_id', 'granted'),
+        ('correct_id', 'incorrect_id', 'granted'),
+        ('correct_id', 'correct_name', 'granted'),
+        ('correct_id', 'incorrect_name', 'granted'),
+        ('correct_id', 'ignore', 'granted'),
+        ('incorrect_id', 'correct_id', 'granted'),
+        ('incorrect_id', 'incorrect_id', 'denied'),
+        ('incorrect_id', 'correct_name', 'granted'),
+        ('incorrect_id', 'incorrect_name', 'denied'),
+        ('incorrect_id', 'ignore', 'denied'),
+        ('correct_name', 'correct_id', 'granted'),
+        ('correct_name', 'incorrect_id', 'granted'),
+        ('correct_name', 'correct_name', 'granted'),
+        ('correct_name', 'incorrect_name', 'granted'),
+        ('correct_name', 'ignore', 'granted'),
+        ('incorrect_name', 'correct_id', 'granted'),
+        ('incorrect_name', 'incorrect_id', 'denied'),
+        ('incorrect_name', 'correct_name', 'granted'),
+        ('incorrect_name', 'incorrect_name', 'denied'),
+        ('incorrect_name', 'ignore', 'denied'),
+        ('ignore', 'correct_id', 'granted'),
+        ('ignore', 'incorrect_id', 'denied'),
+        ('ignore', 'correct_name', 'granted'),
+        ('ignore', 'incorrect_name', 'denied'),
+        ('ignore', 'ignore', 'denied'),
+    ])
+    def test_default_answer_simple_creds_auth(self, meta_uid, meta_gid,
+                                              expected_access):
+
+        params = {'auth_type': AuthPlugin.SimpleCredsAuth,
+                  'meta_uid': meta_uid,
+                  'meta_gid': meta_gid}
+
+        with CustodiaServer(self.test_dir, params) as server:
```

Comment:
I think, the speed is not our issue as long as you could run tests without those. But the point is that I look on Custodia as on black box. I really would like to test it in this way, because it is really essential use case. I want to see that entire Custodia work properly in this point.

If this is unacceptable to you, I can limit the set considerably.